garak/_config.py

```diff
@@ -141,6 +151,11 @@ def _load_yaml_config(settings_filenames) -> dict:
         with open(settings_filename, encoding="utf-8") as settings_file:
             settings = yaml.safe_load(settings_file)
             if settings is not None:
+                if _key_exists(settings, "api_key"):
+                    logging.info(f"API key found in {settings_filename}. Checking readability...")
+                    res = os.stat(settings_filename)
+                    if res.st_mode & stat.S_IROTH or res.st_mode & stat.S_IRGRP:
+                        logging.warn(f"A possibly secret value (`api_key`) was detected in {settings_filename}, which is readable by users other than yourself.")
```

Comment:
You know, to get this surfaced to CLI users, we could add this near the top:

```
import garak.command
```

and then do something like:

```suggestion
                        msg = f"A possibly secret value (`api_key`) was detected in {settings_filename}, which is readable by users other than yourself. Consider changing permissions on this file to only be readable by you."
                        logging.warn(msg)
                        command.hint(msg)
```